### PR TITLE
chore: add the scorecard supply chain security workflow

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This adds the scorecard workflow (and badges) to the repository

## Related Issue

Fixes #N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

**Comment by Racer159** [comment]
(made this a standalone workflow for now but if desired could convert it to pull from uds-common instead)

File: .github/workflows/scorecard.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          persist-credentials: false
+
+      - name: Run analysis
+        uses: ossf/scorecard-action@62b2cac7ed8198b15735ed49ab1e5cf35480ba46 # v2.4.0
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          repo_token: ${{ secrets.SCORECARD_READ_TOKEN }}
```
Comment (mjnagel) [inline .github/workflows/scorecard.yaml]: This token doesn't exist in this repo - is it something that needs to be added/expanded to our repo at the org level?

**Review by mjnagel — COMMENTED** [review]
Just the one question, otherwise no concerns here.

File: .github/workflows/scorecard.yaml (hunk 1 above)
Comment (mjnagel) [reply .github/workflows/scorecard.yaml]:
```suggestion
```

For clarity.

**Review by mjnagel — APPROVED** [review]

**Pull request merged**